tests/unit/parser.js

```diff
@@ -6584,3 +6584,21 @@ exports.functionKeyword = function (test) {
 
   test.done();
 };
+
+exports.nonGeneratorAfterGenerator = function (test) {
+  var run;
+  var code = [
+    'var obj = {',
+    '  *gen() {',
+    '    yield 1;',
+    '  },',
+    '  non_gen() {',
```

Comment:
It's not super-clear what this is testing. Do we normally report an error if we think we have a generator without a yield expression? If so, probably worth marking that in a comment, just to help reviewer-sanity =)